src/detect-engine.h

```diff
@@ -164,7 +164,7 @@ void DetectEngineSetParseMetadata(void);
 void DetectEngineUnsetParseMetadata(void);
 int DetectEngineMustParseMetadata(void);
 
-int DetectBufferSetActiveList(Signature *s, const int list);
+int WARN_UNUSED DetectBufferSetActiveList(Signature *s, const int list);
```

Comment:
Since you've modified the offending places (where the return code was ignored), is `WARN_UNUSED` still needed?

Reply:
Its not needed for the current codebase but is this function not going to be used in future again? If yes, I think check would prove to be helpful at all those places.

Reply:
I agree that this would be useful moving forward.